crates/matrix-sdk-ui/src/room_list_service/state.rs

```diff
@@ -83,6 +127,8 @@ impl State {
             }
         };
 
+        self.set(next_state.clone());
```

Comment:
This is the error that makes `test_sliding_sync_indicator` to fail, and possibly other tests. This method caculates the next state, but the state must be updated if and only if a sync has run. It's `RoomListService::sync` that is responsible to update the state.